src/sentry/issues/endpoints/project_events.py

```diff
@@ -61,18 +66,48 @@ def get(self, request: Request, project: Project) -> Response:
         """
         Return a list of events bound to a project.
         """
+        from rest_framework.exceptions import ParseError
+
         from sentry.api.paginator import GenericOffsetPaginator
 
         query = request.GET.get("query")
         conditions = []
         if query:
             conditions.append([["positionCaseInsensitive", ["message", f"'{query}'"]], "!=", 0])
 
+        try:
+            start, end = get_date_range_from_params(
+                request.GET, optional=True
+            )  # NB: this will always
+            # return timezone-aware datetimes, even if the parameters passed didn't have a timezone (will default to UTC)
+        except InvalidParams:
+            raise ParseError(detail="Invalid date range parameters provided")
+
         event_filter = eventstore.Filter(conditions=conditions, project_ids=[project.id])
+
+        # Add date filtering, potentially combined with date limit feature flag enforcement:
         if features.has(
             "organizations:project-event-date-limit", project.organization, actor=request.user
```

Comment:
@wedamija do you know what this feature flag is for? seems to not be enabled for any orgs